src/Microsoft.Framework.Configuration/ConfigurationBuilder.cs

```diff
@@ -24,7 +25,7 @@ public ConfigurationBuilder(string basePath, params IConfigurationSource[] sourc
                 }
             }
 
-            BasePath = basePath;
+            _properties["BasePath"] = string.Empty;
```

Comment:
See my other comment below about moving the path resolution logic to live along side the `SetPath()` method. This case should be fine because the path resolution logic can default to whatever it needs to (`string.Empty`?) if the property is not set.